examples/gcf-pubsub-vm-delete-event-handler/main.py

```diff
@@ -46,10 +58,59 @@ def setup_logging(context=None):
     # See https://github.com/googleapis/google-api-python-client/issues/299
     api_logger = logging.getLogger('googleapiclient')
     api_logger.setLevel(logging.ERROR)
+    # Structured logs for Stackdriver
+    root_logger = logging.getLogger()
+    # Clear the root logger
+    print({'handlers': root_logger.handlers})
+
+    ## TODO(jmccune): Remove default handlers once permissions are sorted out.
+    # We want to see STDERR warnings for the time being.
+    # default_handlers = [h for h in root_logger.handlers]
+    # [root_logger.removeHandler(h) for h in default_handlers]
+    # print({'handlers': root_logger.handlers})
+
+    # Add the stackdriver handler
+    # TODO(jmccune): Verify the stackdriver handler is well behaved in pytest
+    if __flags.logging_setup_done is False:
+        #print("Setting up logging because logging_setup_done is False")
+        #print('Event ID: {}'.format(context.event_id))
+        #print('Event type: {}'.format(context.event_type))
+        #print("Resource: {}".format(context.resource))
+        #print("Resource Attributes: {}".format(dir(context.resource)))
+        #print("ENVIORNMENT")
+        #keys = [k for k in os.environ.keys()]
+        #keys.sort()
+        #for k in keys:
+        #    print("{}: {}".format(k, os.getenv(k)))
+        client = google.cloud.logging.Client()
+        # res = Resource(
+        #     type = context.event_type,
+        #     labels = {
+        #     }
+        # )
+        stackdriver = CloudLoggingHandler(client)
+        # TODO(jmccune) Replace with g.c.l.handlers.setup_logging()
+        google.cloud.logging.handlers.setup_logging(stackdriver)
+        __flags.logging_setup_done = True
+        #print({'handlers': root_logger.handlers})
+
     # Set level of our logger.
-    logger = logging.getLogger(LOGNAME)
-    logger.setLevel(level)
-    return logger
+    log = logging.getLogger(LOGNAME)
+    log.setLevel(level)
+    return log
+
+
+def stackdriver_logging(message_object: dict, severity: str):
+    """Logs the provided dictionary message_object with Stackdriver as structured data
+    Args:
+        message_object (dict): The structured data to be provided as a log
+            message. Dictionary must contain a 'message' key.
+        severity (string): The severity level to log (e.g. INFO, WARNING, ERROR,
+            CRITICAL...)
+    """
+    logging_client = google.cloud.logging.Client()
+    logger = logging_client.logger('pubsub-vm-delete')
+    logger.log_struct(message_object, severity=severity)
```

Comment:
Need to verify this is synchronous.  I'm concerned async logging could create a race condition based on [Do not start background activities](https://cloud.google.com/functions/docs/bestpractices/tips#do_not_start_background_activities)